transport/controller.cc

```diff
@@ -92,7 +92,11 @@ future<> controller::do_start_server() {
             std::shared_ptr<seastar::tls::credentials_builder> cred;
         };
 
-        std::vector<listen_cfg> configs({{ socket_address{ip, cfg.native_transport_port()}, false }});
+        std::vector<listen_cfg> configs;
+
+        if (cfg.native_transport_port.is_set()) {
```

Comment:
I don't think this is a good idea to drop the default. This is not a backwards compatible change and can break existing deployments. If we decide to go with this, let's at least check `cfg.native_transport_port.is_set() || !cfg.native_transport_port_ssl.is_set()` to disable this only when SSL is enabled.

Reply:
Agree. We can use a port number of zero as a hacky way of disabling it.

Reply:
@avikivity I don't understand what you mean here. If I change the if expression to @haaawk's suggestion, thing will work as expected, and there's no need for the zero port hack, right?

Reply:
I agreed with "I don't think this is a good idea to drop the default".

The trick with the boolean expression is too tricky. It changes a configuration that had 9042 defaulted and 9142 non-defaulted to one that doesn't have 9042 at all.